src/pd/client.rs

```diff
@@ -226,7 +231,15 @@ impl PdClient for RpcClient {
         })?;
         check_resp_header(resp.get_header())?;
 
-        Ok(resp.take_stores().to_vec())
+        let mut stores = resp.take_stores().to_vec();
+        if !include_tombstone {
+            for i in (0..stores.len()).rev() {
+                if stores[i].get_state() == metapb::StoreState::Tombstone {
+                    stores.swap_remove(i);
+                }
+            }
```

Comment:
If we are allowed to use nightly APIs this could be simplified as

```rust
#![feature(drain_filter)]

stores.drain_filter(|s| s.get_state() == metapb::StoreState::Tombstone);
```

However according to rust-lang/rust#43244 there isn't a consensus on the exact behavior, so it would likely stay unstable for quite a long time.

Reply:
I think `swap_remove` is fine. :thinking: